File: src/main/java/com/github/bjoernpetersen/deskbot/JavafxHostServices.java
package com.github.bjoernpetersen.deskbot;

import com.github.bjoernpetersen.jmusicbot.platform.ContextSupplier;
import com.github.bjoernpetersen.jmusicbot.platform.HostServices;
import java.net.URL;
import javax.annotation.Nonnull;

public class JavafxHostServices implements HostServices {

  @Override
  public void openBrowser(URL url) {
    DeskBot.getInstance().getHostServices().showDocument(url.toExternalForm());
  }

  @Nonnull
  @Override
  public ContextSupplier contextSupplier() throws IllegalStateException {
    throw new IllegalStateException("Context not available on Desktop systems");
  }
}

File: src/main/java/com/github/bjoernpetersen/deskbot/view/JavaFxHostServices.kt
package com.github.bjoernpetersen.deskbot.view

import com.github.bjoernpetersen.jmusicbot.platform.ContextSupplier
import com.github.bjoernpetersen.jmusicbot.platform.HostServices
import tornadofx.*
import java.net.URL

class JavaFxHostServices : HostServices {

  override fun openBrowser(url: URL) {
    val configWindow = find(ConfigWindow::class)
    configWindow.hostServices.showDocument(url.toExternalForm())
  }

  @Throws(IllegalStateException::class)
  override fun contextSupplier(): ContextSupplier {
    throw IllegalStateException("Context not available on Desktop systems")
  }
}
